docs/java/features/odata/use-vdm.md

```diff
@@ -466,3 +603,36 @@ POST /ODataService/API_BUSINESS_PARTNER/A_BusinessPartner(123)/to_BusinessPartne
 
 </TabItem>
 </Tabs>
+
+## Migrate to improved OData VDM (Beta)
+The SAP Cloud SDK released an experimental new OData VDM implementation. It improves performance of your Java apps by avoiding unnecessary OData metadata calls and thereby saving roundtrips to the backend system. Also, the Cloud SDK team implemented an OData client library that serves both OData V2 and V4 protocols. Consequently, this approach allows for faster delivery of new features in the future. Note that this new implementation is in Beta state. We'll announce the production readiness explicitly in our release channels.
+
+The following steps outline how to migrate your existing code to leverage the improved OData VDM implementation.
+
+:::note
+We  distinguish the operation types _Create_, _Delete_, _Update_, _Get All Entities_ and _Get Entity By Key_.
+:::
+- For any operation type, replace any call to `execute` with `executeRequest`.
+```java
+//Current OData VDM 
+service.createBusinessPartner(partner).execute(destination);
+
+//Improved OData VDM
+service.createBusinessPartner(partner).executeRequest(destination);
+```
+- For _Create_ and _Update_ operations, call `getModifiedEntity()` to obtain the created entity representation.
+```java
+//Current OData VDM - Create
+final BusinessPartner partner = service.createBusinessPartner(partner).execute(destination);
+//Current OData VDM - Update
+final BusinessPartner partner = service.updateBusinessPartner(partner).execute(destination);
+
+//Improved OData VDM - Create
+final BusinessPartner partner = service.createBusinessPartner(partner).executeRequest(destination).getModifiedEntity();
+//Improved OData VDM - Update
+final BusinessPartner partner = service.updateBusinessPartner(partner).executeRequest(destination).getModifiedEntity();
+```
+- For _Delete_, _Get All Entities_ and _Get Entity By Key_ operations you do **not** need to call an additional method after `executeRequest`.
+- Remove any call to `cachingMetadata()` and `withoutCachingMetadata()` as they do not have any effect any longer.
```

Comment:
I would weaken this to `you can remove it if you like, it's obsolete`

Reply:
same goes for the error handler

Reply:
Can't do that for the error handler as it is not obsolete, it is just currently not yet supported. I assume we plan supporting it in the future.